t/plugin-security-extended-status.t

```diff
@@ -0,0 +1,150 @@
+use Mojo::Base -strict;
+use Test::Mojo;
+use Test::More;
+
+use Mojolicious::Lite;
+
+get '/securitytest' => sub {
+  my $c = shift->openapi->valid_input or return;
+  $c->render(openapi => {message => "ok"}, status => 200);
+  },
+  'securitytest';
+
+our %checks;
+plugin OpenAPI => {
+  url      => 'data://main/sec.yaml',
+  schema   => 'v3',
+  security => {
+    api_key => sub {
+      my ($self, $definition, $scopes, $cb) = @_;
+      my $apikey  = $self->req->headers->header('apikey');
+      my $apiuser = $self->req->headers->header('apiuser');
+      return $self->$cb("apikey/apiuser missing") unless $apikey and $apiuser;
+      if ($apikey eq "authenticated" and $apiuser eq "authenticated") {
+        $self->stash(status => 403);
+        return $self->$cb("Permission denied");
+      }
+      if ($apikey eq "authorized" and $apiuser eq "authorized") {
+        return $self->$cb();
+      }
+      return $self->$cb("Unauthorized");
+    },
+  },
+};
+
+my $t = Test::Mojo->new;
+
+{
+  local %checks;
```

Comment:
You don't need `%checks` and you can drop the blocks+indentation.

Reply:
Done